File: src/uis/ReactWeb/src/entities/rent/models/RentProduct.ts
import {User} from '@/entities/user/models/User.ts';
import {Product} from '@/entities/rent/models/Product.ts';

export interface RentProduct {
    id?:string;
    userId:string;
    user?:User;
    productId:string;
    product?:Product;
    fromDate?:Date;
    toDate?:Date;
    status?:string;
}

File: src/services/Hunting/Modules/Rent/Rent.Domain/Entities/RentProduct.cs
using Rent.Domain.Enums;
using Rent.Domain.Interfaces;

namespace Rent.Domain.Entities
{
    public class RentProduct
        : IBaseEntity
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; } = null!;
        public Guid ProductId { get; set; }
        public Product Product { get; set; } = null!;
        public DateTime FromDate { get; set; }
        public DateTime ToDate { get; set; }
        public RentStatus Status { get; set; }
    }
}
